extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -711,3 +711,24 @@ export async function fileContainsCreateTableStatement(fullPath: string, project
 
 	return containsCreateTableStatement;
 }
+
+/**
+ * Gets target platform based on the server edition/version
+ * @param profile server connection profile
+ * @returns target platform for the database project
+ */
+export async function getTargetPlatformFromServerVersion(profile: azdataType.IConnectionProfile): Promise<SqlTargetPlatform | undefined> {
+	const serverInfo = await azdataType.connection.getServerInfo(profile.id);
+	const isCloud = serverInfo.isCloud;
+
+	let targetPlatform;
+	if (isCloud) {
+		const engineEdition = serverInfo.engineEditionId;
+		targetPlatform = engineEdition === azdataType.DatabaseEngineEdition.SqlDataWarehouse ? SqlTargetPlatform.sqlDW : (engineEdition === azdataType.DatabaseEngineEdition.SqlDatabase ? SqlTargetPlatform.sqlAzure : undefined);
```

Comment:
if it isn't DW, can we just assume it's sqlAzure and skip the check if the engineEditoin is SqlDatabase?

Reply:
Projects support Sql Edge as well (https://github.com/microsoft/azuredatastudio/blob/5d30c5eb88d2386bf4d822fa95098ec8f47e3eca/extensions/sql-database-projects/src/sqldbproj.d.ts). So I thought setting it to SQL Azure wouldn't be completely correct, hence the extra check. But if you think it should safe to skip the additional check, I can do that to.

Reply:
I think it should be ok to set the target platform to sqlAzure if it's cloud and not DW. @Benjin any concerns with this? I don't think there are that many people creating projects from edge dbs, and the new edge project template creates projects with the target platform 150.

Reply:
Updated it for now, waiting on Benjin for confirmation.